env.go

```diff
@@ -391,6 +409,38 @@ func doParseField(
 	return nil
 }
 
+func isMapOfStructs(refTypeField reflect.StructField, opts Options) bool {
+	field := refTypeField.Type
+	if reflect.Ptr == field.Kind() {
+		field = field.Elem()
+	}
+
+	if reflect.Map != field.Kind() {
+		return false
+	}
+
+	field = field.Elem()
+
+	if reflect.Ptr == field.Kind() {
+		field = field.Elem()
+	}
+
+	_, ignore := defaultBuiltInParsers[field.Kind()]
+
+	if !ignore {
+		_, ignore = opts.FuncMap[field]
+	}
+
+	if !ignore {
+		_, ignore = reflect.New(field).Interface().(encoding.TextUnmarshaler)
+	}
+
+	if !ignore {
+		ignore = reflect.Struct != field.Kind()
+	}
+	return !ignore
```

Comment:
What about this?

```suggestion
	_, ignore := defaultBuiltInParsers[field.Kind()]
	if ignore {
		return false
	}

	_, ignore = opts.FuncMap[field]
	if ignore {
		return false
	}

	_, ignore = reflect.New(field).Interface().(encoding.TextUnmarshaler)
	if ignore {
		return false
	}

	ignore = reflect.Struct != field.Kind()
	return !ignore
```

It would be clearer, no?

I hope I didn't invert the logic

Reply:
OK, I can understand then.

I would fix the code you add in this PR, but why not.

Maybe fixing the other method in another PR first, or merge this, and open another PR  to fix the style.

But OK, it's out of the PR scope maybe.